contract/contracts/PixelDevsUkraineDonation.sol

```diff
@@ -49,42 +48,68 @@ contract PixelDevsUkraineDonation is
         baseURI = _newBaseURI;
     }
 
-    function setMinimumMintPrice(uint256 _newPrice) public onlyOwner {
-        // Mint price in wei
-        emit MinimumMintPriceUpdated(minimumMintPrice, _newPrice);
-        minimumMintPrice = _newPrice;
+    function setTierPricing(
+        uint256 steel,
+        uint256 bronze,
+        uint256 silver,
+        uint256 gold,
+        uint256 diamond,
+        uint256 platinum
+    ) public onlyOwner {
+        emit TierPricingUpdated("steel", tiers["steel"], steel);
+        emit TierPricingUpdated("bronze", tiers["bronze"], bronze);
+        emit TierPricingUpdated("silver", tiers["silver"], silver);
+        emit TierPricingUpdated("gold", tiers["gold"], gold);
+        emit TierPricingUpdated("diamond", tiers["diamond"], diamond);
+        emit TierPricingUpdated("platinum", tiers["platinum"], platinum);
+
+        tiers["steel"] = steel;
+        tiers["bronze"] = bronze;
+        tiers["silver"] = silver;
+        tiers["gold"] = gold;
+        tiers["diamond"] = diamond;
+        tiers["platinum"] = platinum;
     }
 
     function mint() public payable nonReentrant {
-        require(minimumMintPrice <= msg.value, "Not enough MATIC sent");
-        require(contractState, "Contract must be active to mint");
+        require(
+            msg.value >= tiers["steel"],
+            "Please send enough MATIC to meet the minimum threshold of a tier."
+        );
+        require(
+            contractActive,
+            "This contract has been deactivated by the owner and does not currently accept anymore donations."
+        );
 
         _tokenIds.increment();
-        uint256 newTokenId = _tokenIds.current();
-        _safeMint(msg.sender, newTokenId);
-
-        string memory donationType = "platinum";
-
-        if (msg.value <= 25 ether) {
-            donationType = "bronze";
-        } else if (msg.value <= 50 ether) {
-            donationType = "silver";
-        } else if (msg.value <= 100 ether) {
-            donationType = "gold";
-        } else if (msg.value <= 500 ether) {
-            donationType = "diamond";
+        uint256 _token = _tokenIds.current();
+        _safeMint(msg.sender, _token);
+
+        string memory _tier = "steel";
+
+        if (msg.value >= tiers["platinum"]) {
+            _tier = "platinum";
+        } else if (msg.value >= tiers["diamond"]) {
+            _tier = "diamond";
+        } else if (msg.value >= tiers["gold"]) {
+            _tier = "gold";
+        } else if (msg.value >= tiers["silver"]) {
+            _tier = "silver";
+        } else if (msg.value >= tiers["bronze"]) {
+            _tier = "bronze";
         }
 
-        emit LogTokenMinted(msg.sender, newTokenId, donationType);
-        _setTokenURI(newTokenId, donationType);
+        emit LogTokenMinted(msg.sender, _token, _tier);
+
+        _setTokenURI(_token, string(bytes.concat(bytes(_tier), bytes(".png"))));
```

Comment:
Good point -- probably better to just leave it simple since this is a one-shot deal and we don't necessarily need metadata.

Reply:
Actually thanks for pointing this out. Come to think of it this is probably a violation of the ERC-721 standard and I'm not sure NFTs would render on OpenSea? It was my first time uploading assets on Piñata so that's why I missed af step.

We can either

- Have someone who knows what they're doing upload the assets once we have them (probably a good idea anyway) and revert this change
- Use an approach like https://github.com/Developer-DAO/developer-dao-nft-contract/blob/c65fd8a8712e328f34c0b8d1fb7e4c778eca1185/hh/contracts/Dev.sol#L1516 to simply embed the json straight in the contract given it's just description, image and tier fields needed. This would save a roundtrip to IPFS.

Reply:
Ah, good point about breaking ERC-721 standard. At least we only have 6 pieces of metadata.

Thanks also for looking into Piñata and taking IPFS on!